docs/extending/editor_api.md

```diff
@@ -0,0 +1,168 @@
+const content = await previewController?.extractContent();
+await previewController?.runContentChecks();
+```
+
+## Example: generating meta description with an LLM
+
+Extracting the previewed content using the `PreviewController` can be useful for different use cases. One example is generating a meta description for the page using a Large Language Model (LLM). The following example demonstrates a [custom Stimulus controller](extending_client_side_stimulus) that uses an LLM from the browser's [Summarizer API](https://developer.mozilla.org/en-US/docs/Web/API/Summarizer) to generate the page's meta description.
+
+```javascript
+/* js/summarize.js */
+
+class SummarizeController extends window.StimulusModule.Controller {
+  static targets = ['suggest'];
+
+  static values = {
+    input: { default: '', type: String },
+  };
+
+  /** Only load the controller if the browser supports the Summarizer API. */
+  static get shouldLoad() {
+    return 'Summarizer' in window;
+  }
+
+  /** The previewed content's language. */
+  contentLanguage = document.documentElement.lang || 'en';
+  /** A cached Summarizer instance Promise to avoid recreating it unnecessarily. */
+  #summarizer = null;
+
+  /** Promise of a browser Summarizer instance. */
+  get summarizer() {
+    if (this.#summarizer) return this.#summarizer; // Return from cache
+    this.#summarizer = Summarizer.create({
+      // Change the Summarizer's configuration as needed
+      sharedContext: `A summary of a webpage's content, suitable for use as a meta description.`,
+      type: 'teaser',
+      length: 'short',
+      format: 'plain-text',
+      expectedInputLanguages: [this.contentLanguage],
+      outputLanguage: document.documentElement.lang,
+    });
+    return this.#summarizer;
+  }
+
+  connect() {
+    this.input = this.element.querySelector(this.inputValue);
+    this.renderFurniture();
+  }
+
+  renderFurniture() {
+    const prefix = this.element.closest('[id]').id;
+    const buttonId = `${prefix}-generate`;
+    const button = /* html */ `
+      <button
+        id="${buttonId}"
+        type="button"
+        data-summarize-target="suggest"
+        data-action="summarize#generate"
+        class="button"
+      >
+        Generate suggestions
+      </button>
+    `;
+    this.element.insertAdjacentHTML('beforeend', button);
+
+    this.outputArea = document.createElement('div');
+    this.element.append(this.outputArea);
+  }
+
+  renderSuggestion(suggestion) {
+    const template = document.createElement('template');
+    template.innerHTML = /* html */ `
+      <div>
+        <output for="${this.suggestTarget.id}">${suggestion}</output>
+        <button class="button button-small" type="button" data-action="summarize#useSuggestion">Use</button>
+      </div>
+    `;
+    this.outputArea.append(template.content.firstElementChild);
+  }
+
+  useSuggestion(event) {
+    this.input.value = event.target.previousElementSibling.textContent;
```

Comment:
Once Matthew's client-side `EditHandler` docs are in, we should probably update this to use that mechanism rather than working with the HTML input directly.